velox/exec/tests/utils/TestIndexStorageConnector.h

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/HashTable.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox::exec::test {
+
+inline static const std::string kTestIndexConnectorName("IndexConnector");
+
+/// The index table used for index lookup storage test. The data is stored in an
+/// in-memory hash table, and the lookup is performed on the hash table using
+/// probe and list join result similar as the hash probe operator does.
+struct TestIndexTable {
+  RowTypePtr keyType;
+  RowTypePtr dataType;
+  std::shared_ptr<BaseHashTable> table;
+
+  TestIndexTable(
+      RowTypePtr _keyType,
+      RowTypePtr _dataType,
+      std::shared_ptr<BaseHashTable> _table)
+      : keyType(std::move(_keyType)),
+        dataType(std::move(_dataType)),
+        table(std::move(_table)) {}
+};
+
+/// The index table handle which provides the index table for index lookup.
+class TestIndexTableHandle : public connector::ConnectorTableHandle {
+ public:
+  explicit TestIndexTableHandle(
+      std::string connectorId,
+      std::shared_ptr<TestIndexTable> indexTable,
+      bool asyncLookup)
+      : ConnectorTableHandle(std::move(connectorId)),
+        indexTable_(std::move(indexTable)),
+        asyncLookup_(asyncLookup) {}
+
+  ~TestIndexTableHandle() override = default;
+
+  std::string toString() const override {
+    return fmt::format(
+        "IndexTableHandle: num of rows: {}, asyncLookup: {}",
+        indexTable_ ? indexTable_->table->rows()->numRows() : 0,
+        asyncLookup_);
+  }
+
+  const std::string& name() const override {
+    static const std::string kTableHandleName{"TestIndexTableHandle"};
+    return kTableHandleName;
+  }
+
+  bool supportsIndexLookup() const override {
+    return true;
+  }
+
+  folly::dynamic serialize() const override {
+    folly::dynamic obj = folly::dynamic::object;
+    obj["name"] = name();
+    obj["connectorId"] = connectorId();
+    obj["asyncLookup"] = asyncLookup_;
+    return obj;
+  }
+
+  static std::shared_ptr<TestIndexTableHandle> create(
+      const folly::dynamic& obj,
+      void* context) {
+    // NOTE: this is only for testing purpose so we don't support to serialize
```

Comment:
If we don't expect this code to run, why can't we just throw here and in serialize?